SAMD51 MicroMod variant name doesn't match directory name

The variant configuration for the SAMD51 MicroMod is incorrect in `boards.txt` which results in compilation errors. The config specifies `SparkFun SAMD51 MicroMod` while the variant directory is `SparkFun_SAMD51_MicroMod`.

```
In file included from /Users/$USER/Library/Arduino15/packages/SparkFun/hardware/samd/1.7.6/cores/arduino51/Arduino.h:81:0,
                 from /var/folders/wy/vc_834_n67j2fj8xqnqk0n4r0000gn/T/arduino_build_747408/sketch/Blink.ino.cpp:1:
/Users/$USER/Library/Arduino15/packages/SparkFun/hardware/samd/1.7.6/cores/arduino51/delay.h:24:21: fatal error: variant.h: No such file or directory
 #include "variant.h"
                     ^
compilation terminated.
exit status 1
Error compiling for board SparkFun SAMD51 MicroMod.
```

--- a/sparkfun/samd/boards.txt
+++ b/sparkfun/samd/boards.txt
@@ -296,7 +296,7 @@ micromod_samd51.build.core=arduino51
 micromod_samd51.build.extra_flags=-D__SAMD51J20A__ -D__SAMD51__ {build.usb_flags} -D__FPU_PRESENT -DARM_MATH_CM4 -mfloat-abi=hard -mfpu=fpv4-sp-d16
 micromod_samd51.build.ldscript=linker_scripts/gcc/flash_with_bootloader.ld
 micromod_samd51.build.openocdscript=openocd_scripts/arduino_zero.cfg
-micromod_samd51.build.variant=SparkFun SAMD51 MicroMod
+micromod_samd51.build.variant=SparkFun_SAMD51_MicroMod
 micromod_samd51.build.variant_system_lib=
 micromod_samd51.build.vid=0x1B4F
 micromod_samd51.build.pid=0xF020

